lib/main.js

```diff
@@ -166,7 +179,16 @@ app.use(connect_util.filterByRegex(
     // jcheng 8/16/2013: Having trouble chaining error404 handler with
     // the asset handler. Just check explicitly for now, since the assets
     // dir only contains a single file right now.
-    if (req.url === 'sockjs-0.3.min.js' || req.url === 'shiny-server.js' || req.url === 'shiny-server.css')
+    if (req.url === 'shiny-server.js'){
+      res.setHeader('Content-Type', 'application/javascript');
+
+      // Disable cache so server-side updates to available protocols show immediately.
+      res.setHeader('Cache-Control', 'no-cache, no-store, must-revalidate');
+      res.setHeader('Pragma', 'no-cache');
+      res.setHeader('Expires', '0');
+
+      res.end(ssjs({protocols: availableProtocolsStr}));
```

Comment:
There's no reason to render the template per request; the input data remains constant. Why not render the template once after processing the config?